src/libprojectM/Renderer/BeatDetect.cpp

```diff
@@ -40,17 +40,10 @@
 BeatDetect::BeatDetect(PCM* _pcm)
     : pcm(_pcm)
 {
-    for (unsigned y = 0; y < this->vol_buffer.size(); y++)
-        this->vol_buffer[y] = 0;
-
-    for (unsigned y = 0; y < this->bass_buffer.size(); y++)
-        this->bass_buffer[y] = 0;
-
-    for (unsigned y = 0; y < this->mid_buffer.size(); y++)
-        this->mid_buffer[y] = 0;
-
-    for (unsigned y = 0; y < this->treb_buffer.size(); y++)
-        this->treb_buffer[y] = 0;
+    this->vol_buffer.fill(0);
```

Comment:
Good catch. Is `{{0.f}}` or `{0.f}` clearer? I'm leaning towards the former.

Reply:
The former is two braces, I think it makes more sense because it initializes the `std::array` with an array, which is initialized with `{0.f}`, which becomes `std::array{{0.f}}`. However, following that rule, you would initialize 2d arrays with `std::array{{{{0.f}}}}`, (initialize inner array of `std::array` of inner array of `std::array`) which is *quite* bulky :^). I do think keeping the `0.f` is clearer than omitting it however.

Reply:
Sure, so it'd be:
```c++
std::array<float, BEAT_HISTORY_LENGTH> vol_buffer{{0.f}};
```